readthedocs/rtd_tests/tests/test_oauth.py

```diff
@@ -1561,6 +1602,38 @@ def test_send_build_status_successful(self, session, mock_logger, mock_structlog
             "GitHub commit status created for project.",
         )
 
+    @mock.patch("readthedocs.oauth.services.github.structlog")
+    @mock.patch("readthedocs.oauth.services.github.log")
+    @mock.patch("readthedocs.oauth.services.github.GitHubService.session")
+    def test_send_build_status_success_when_not_built(self, session, mock_logger, mock_structlog):
```

Comment:
You shouldn't need to mock the log calls, since you are already mocking `session`, you should be able to inspect the call in session.post. Alternatively, you can use requests_mock.